### PR TITLE
Add contributing guidelines

### PR DESCRIPTION
**Pull request by appgurueu**
Closes https://github.com/TheAlgorithms/Algorithms-Explanation/issues/187.

Up for discussion: What am I missing here?

Could probably use good/bad examples, as well as a note to check grammar/spelling/caps.

@Panquesito7 do you have Markdown conventions to add?

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+### Markdown Conventions
+
+[GitHub-flavored Markdown is used](https://github.github.com/gfm/). Explanations should render well when viewed from GitHub.
+
+* **Do not add redundant formatting.** Formatting should always be meaningful.
+  If you apply a certain formatting to all elements of a certain kind (e.g. adding emphasis around all headings), you're doing something wrong.
+* Use ATX-style "hashtag" headings: `#`, `##`, `###`, `####`, `#####`, `######` rather than Setext-style "underline" headings.
+  Leave blank lines around headings.
+* Indent lists by two blanks. Prefer `*` over `-` for bulleted lists. Enumerate numbered lists correctly, starting at `1`.
```
Comment (Panquesito7) [inline CONTRIBUTING.md:55]: I believe we should use `-` rather than `*`, IMO.
Not sure what is the best option, though.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+* Add a space before an opening parenthesis `(`. Do not add a space before the closing parenthesis `)`.
+* Add spaces around (but not inside) quotes. Use single quotes for quotes within quotes.
+
+### Markdown Conventions
+
+[GitHub-flavored Markdown is used](https://github.github.com/gfm/). Explanations should render well when viewed from GitHub.
+
+* **Do not add redundant formatting.** Formatting should always be meaningful.
+  If you apply a certain formatting to all elements of a certain kind (e.g. adding emphasis around all headings), you're doing something wrong.
+* Use ATX-style "hashtag" headings: `#`, `##`, `###`, `####`, `#####`, `######` rather than Setext-style "underline" headings.
```
Comment (Panquesito7) [inline CONTRIBUTING.md:53]: Should we mention the first line should _always_ be a `#` header?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+Do not change the structure or formatting of the original (English) explanation when translating or updating translations.
+You may change number formatting to fit the locale
+(`42.33` may be formatted as `42,33` in a german translation for instance)
+as long as it doesn't create ambiguities
+(e.g. `[42,33, 13]` in an array would be disallowed and should either remain `[42.33, 13]` or use a different delimiter `[42,33; 13]`).
+
+## Writing Explanations
+
+### Structure
```
Comment (Panquesito7) [inline CONTRIBUTING.md:21]: To make it easier for people to add an algorithm explanation, we could add a typical structure of how it should look. We can remove it from the root `README.md` of each directory.

File: CONTRIBUTING.md (hunk 1 above)
Comment (appgurueu) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 2 above)
Comment (appgurueu) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+### Markdown Conventions
+
+[GitHub-flavored Markdown is used](https://github.github.com/gfm/). Explanations should render well when viewed from GitHub.
+
+* **Do not add redundant formatting.** Formatting should always be meaningful.
+  If you apply a certain formatting to all elements of a certain kind (e.g. adding emphasis around all headings), you're doing something wrong.
+* Use ATX-style "hashtag" headings: `#`, `##`, `###`, `####`, `#####`, `######` rather than Setext-style "underline" headings.
+  Leave blank lines around headings.
+* Indent lists by two blanks. Prefer `*` over `-` for bulleted lists. Enumerate numbered lists correctly, starting at `1`.
+* Use fenced code blocks (and specify the correct language) rather than using indented code blocks.
```
[the remark this answers is not in the document]
Comment (appgurueu) [reply CONTRIBUTING.md]: Why `txt`? Leaving it blank should be fine.

[the remark this answers is not in the document]
Comment (Panquesito7) [reply CONTRIBUTING.md]: > Why `txt`? Leaving it blank should be fine.

This prevents a Markdown Linter warning. It's not 100% necessary, but will make it cleaner, IMO.

**Comment by Panquesito7** [comment]
Any updates on this? Thanks.

File: CONTRIBUTING.md (hunk 3 above)
Comment (appgurueu) [reply CONTRIBUTING.md:21]: I'd use #201 for that.

File: CONTRIBUTING.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (appgurueu) [reply CONTRIBUTING.md]: Which Markdown linter?

[the remark this answers is not in the document]
Comment (Panquesito7) [reply CONTRIBUTING.md]: I'm using this one from VSCode marketplace: https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint

File: CONTRIBUTING.md (hunk 3 above)
Comment (Panquesito7) [reply CONTRIBUTING.md:21]: We can use that. 🙂

**Review by Panquesito7 — DISMISSED** [review]
Looks good once https://github.com/TheAlgorithms/Algorithms-Explanation/pull/199#discussion_r1003626028 is resolved.
Thanks, though! 🚀

**Review by Panquesito7 — APPROVED** [review]
Awesome. Thanks! 🎉

**Pull request merged**